internal/common/function/function.go

```diff
@@ -661,3 +682,34 @@ func ComputePrefixForObjectStorage(namespace, customPrefix string) string {
 	}
 	return namespace
 }
+
+// HasBSLChangesForApproval checks if there are any changes between the old and new BackupStorageLocationSpec
+// that would require admin approval. We do not want the admin to re-approve the BSL if it's just a matter of
+// the BSL creds being updated. In case of nil specs, we return true to trigger approval.
+func HasBSLChangesForApproval(oldSpec, newSpec *velerov1.BackupStorageLocationSpec) bool {
```

Comment:
is this used?

Reply:
Nice catch. It was firstly implemented when we had the update of NaBSL operation included in the design, however after talking to @shubham-pampattiwar and @weshayutin the idea changed to make things simpler and not allow update of the NaBSL and it's re-approval workflow.

Instead as in the updated design doc, the re-approval requires user to firstly delete NaBSL and then re-create new NaBSL with updated spec. It should be not a very often task to update NaBSLs by the users.

I've removed that function and relevant tests.